File: genetic-packer/src/main/java/genetic/packer/evolution/fitness/impl/TotalFitnessTranslator.java
package genetic.packer.evolution.fitness.impl;

import genetic.packer.evolution.fitness.FitnessComponents;
import genetic.packer.evolution.fitness.FitnessTranslator;
import org.springframework.stereotype.Component;

/**
 * @author piotr.larysz
 */
@Component
public class TotalFitnessTranslator extends BaseFitnessTranslator implements FitnessTranslator<FitnessComponents, Double> {

    @Override
    public Double apply(FitnessComponents fitnessComponents) {
        return this.rateIntersections(fitnessComponents.getIntersections()) + 1 / (fitnessComponents.getVolume() * 100);
    }
}

File: genetic-packer/src/main/java/genetic/packer/evolution/fitness/impl/TotalFitnessTranslator.kt
package genetic.packer.evolution.fitness.impl

import genetic.api.fitness.FitnessTranslator
import genetic.packer.evolution.fitness.FitnessComponents

class TotalFitnessTranslator : BaseFitnessTranslator(),
    FitnessTranslator<FitnessComponents, Double> {

  override fun apply(fitnessComponents: FitnessComponents) =
      rateIntersections(fitnessComponents.intersections) + 1 / (fitnessComponents.volume * 100)
}